src/test/scala/vsys/blockchain/state/contract/vswap/ExecuteVSwapInvalidDiffTest.scala

```diff
@@ -0,0 +1,390 @@
+  property("withdraw liquidity tokens more than depositing in vswap contract") {
+    forAll(preconditionsAndVSwapDepositAndWithdrawLiquidityTokens) { case (genesis: GenesisTransaction, genesis2: GenesisTransaction, regLiquidity: RegisterContractTransaction, regVSwap: RegisterContractTransaction, issueLiquidity: ExecuteContractFunctionTransaction,
+    depositLiquidity: ExecuteContractFunctionTransaction, withdrawLiquidity: ExecuteContractFunctionTransaction, withdrawInvalidLiquidity: ExecuteContractFunctionTransaction) =>
+      assertDiffEi(Seq(TestBlock.create(genesis.timestamp, Seq(genesis, genesis2)), TestBlock.create(depositLiquidity.timestamp, Seq(regLiquidity, regVSwap, issueLiquidity, depositLiquidity))),
+        TestBlock.createWithTxStatus(withdrawLiquidity.timestamp, Seq(withdrawLiquidity), TransactionStatus.Success)) { (blockDiffEi) =>
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.Success
+      }
+      assertDiffEi(Seq(TestBlock.create(genesis.timestamp, Seq(genesis, genesis2)), TestBlock.create(depositLiquidity.timestamp, Seq(regLiquidity, regVSwap, issueLiquidity, depositLiquidity))),
+        TestBlock.createWithTxStatus(withdrawInvalidLiquidity.timestamp, Seq(withdrawInvalidLiquidity), TransactionStatus.ContractTokenBalanceInsufficient)) { (blockDiffEi) =>
+        blockDiffEi.explicitGet().txsDiff.contractNumDB.isEmpty shouldBe true
+        blockDiffEi.explicitGet().txsDiff.portfolios.isEmpty shouldBe false
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.ContractTokenBalanceInsufficient
+      }
+    }
+  }
+
+  val preconditionsAndVSwapSetSwap: Gen[(GenesisTransaction, GenesisTransaction, RegisterContractTransaction,
+    RegisterContractTransaction, RegisterContractTransaction, RegisterContractTransaction, ExecuteContractFunctionTransaction,
+    ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction,
+    ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction)] = for {
+    (genesis, genesis2, master, _, regTokenA, regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA, depositB, depositLiquidity, fee, ts, attach) <-
+      createABLiquidityTokenAndInitVSwap(1000, 1, 1000, 1000, 1, 1000, 1000, 1000,9, 100, 100)
+
+    setSwap <- setSwapVSwapGen(master, regVSwap.contractId, 10L, 10L, attach, fee + 10000000000L, ts)
+    setInvalidSwap <- setSwapVSwapGen(master, regVSwap.contractId, 1000L, 10L, attach, fee + 10000000000L, ts)
+  } yield (genesis, genesis2, regTokenA, regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA,
+    depositB, depositLiquidity, setSwap, setInvalidSwap)
+
+  // set swap token A more than deposit, set swap token B more than deposit has been local tested successfully
+  property("set swap more than deposit") {
```

Comment:
In the set swap section, please add cases in which the initial liquidity is less than minimum.